tap_snapchat_ads/streams.py

```diff
@@ -205,6 +211,11 @@ def process_records(self, catalog, #pylint: disable=too-many-branches
                         self.write_record(stream_name, transformed_record, time_extracted=time_extracted)
                         counter.increment()
 
+                    if stream_name == "ad_accounts" and "timezone" not in record:
+                        # Write the timezone value to original record if it's not selected by the user
+                        record["timezone"] = ad_account_time_zone
```

Comment:
Instead of this line, you can directly write 
```
record["timezone"] = record.get("timezone", None)
```
and remove below lines which are not required :
```
                if stream_name == "ad_accounts":
                    ad_account_time_zone = record.get("timezone", None)
```

Reply:
```
def process_record():
    record = {"timezone": "UTC", "id": "1234"}
    transformed_record = transform(record)
    print(transformed_record)
    print(record)


def transform(data):
    data.pop("timezone")
    return data


process_record()
```

> output:

```
{'id': '1234'}
{'id': '1234'}
```

Reply:
If you do `transformed_record = transform(dict(record))`, we should be able to do it. That way `transform` doesn't modify the original record.

Reply:
thanks @bryantgray, that worked 👍